scripts/apps/archive/controllers/UploadController.ts

```diff
@@ -245,16 +246,15 @@ export function UploadController(
             return false;
         }
 
-        let acceptedFiles: Array<{file: File, getThumbnail: (file: File) => Promise<string>}> = [];
+        let acceptedFiles: Array<{ file: File, getThumbnail: (file: File) => Promise<string> }> = [];
         let uploadOfDisallowedFileTypesAttempted: boolean = false;
+        const imageFiles = [];
```

Comment:
Can you use `acceptedFiles` array and avoid adding `imageFiles`?

There's a problem with this approach, that images are iterated synchronously, but making it async is a cleaner solution than adding additional arrays.

It would also be good to implement it in a way that if some images are too small, the ones which aren't would get uploaded and a message would pop up with the list of filenames of images that weren't uploaded and all their dimensions in addition to required dimensions that are currently displayed.

Here's some psuedo-code on converting the iteration to be async:

```
Promise.all(files.map((file) => {
    if (isImage) {
        if (!imagesAllowed) {
            return Promise.resolve(`Images aren't allowed`);
        }

        return fileDimensionsValid(file).then((dimensions) => {
            if (dimensions.valid) {
                return {
                    file: files,
                    getThumbnail: () => Promise.resolve('<i class="icon--2x icon-video"></i>'),
                };
            } else {
                return {error: 'dimensions not valid', dimensions: dimensions};
            }
        });
    } else {
        // ...
    }
}).then((result) => {
    // add files without errors to acceptedFiles

    // upload

    // display error messages
});

```